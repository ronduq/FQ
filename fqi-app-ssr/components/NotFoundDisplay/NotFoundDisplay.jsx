import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import s from './NotFoundDisplay.module.scss';
import Icon from '../Icon/Icon';
import t from '../Icon/Icon.module.scss';

const NotFoundDisplay = ({classname, type, title, text}) => {
  return (
    <div className={classNames(s.notFoundDisplay, classname)}>
      <div className={s.notFoundDisplayImg}>
        <Icon className={classNames(s.icon, t.iconHuge, t.iconTeal)} iconName={type} />
      </div>
      <p className={s.notFoundDisplayTitle}>{title}</p>
      <p className={s.notFoundDisplayText}>{text}</p>
    </div>
  );
};

NotFoundDisplay.defaultProps = {
  type: 'apple',
  title: 'Not found',
  text: 'No results',
};

NotFoundDisplay.propTypes = {
  classname: PropTypes.string,
  type: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string,
};

export default NotFoundDisplay;
