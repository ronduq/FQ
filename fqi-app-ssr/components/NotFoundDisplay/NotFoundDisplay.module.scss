@import '../../style/settings/colours';
@import '../../style/settings/fonts';

.notFoundDisplay {
  text-align: center;
}
.notFoundDisplayImg {
  height: 280px;
  width: 280px;
  border-radius: 50%;
  background: rgba(2,145,127,0.07);
  margin: 120px auto 20px;
  position: relative;

  .icon {
    position: relative;
    top: 50%;
    transform: translateY(-50%);
  }
}

.notFoundDisplayTitle {
  @include font-family('proxima-bold');
  @include font-size('xlarge');
}

.notFoundDisplayText {
  @include font-size('large');
}

