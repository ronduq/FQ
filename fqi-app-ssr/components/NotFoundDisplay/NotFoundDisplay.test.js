/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import NotFoundDisplay from './NotFoundDisplay'


describe('With Snapshot Testing', () => {
  it('NotFoundDisplay snapshot"', () => {
    const component = renderer.create(<NotFoundDisplay />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  })
})