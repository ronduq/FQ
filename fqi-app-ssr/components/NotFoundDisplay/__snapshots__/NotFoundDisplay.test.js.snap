// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing NotFoundDisplay snapshot" 1`] = `
<div
  className="notFoundDisplay"
>
  <div
    className="notFoundDisplayImg"
  >
    <svg
      aria-hidden="true"
      className="icon iconHuge iconTeal"
      focusable="false"
    >
      <use
        xlinkHref="#apple"
      />
    </svg>
  </div>
  <p
    className="notFoundDisplayTitle"
  >
    Not found
  </p>
  <p
    className="notFoundDisplayText"
  >
    No results
  </p>
</div>
`;
